### PR TITLE
Add pagination to variations list

### LINKED ISSUE #35788: Add pagination to variations list
Variations list should include pagination when more than 5 variations exist.

Variations should be loaded in batches instead of all at once for performance reasons.  The CRUD data store should allow us to get the total from the initial request.  See https://github.com/woocommerce/woocommerce/pull/34613 for usage details.

**Design**

BaoQ3PGPnhAiJCsJtxgej3-fi-106%3A9597&t=Jsh233onKmJpJYtN-0

Image

**Acceptance criteria**

* [ ] Show the pagination when more than 5 variations exist
* [ ] Pagination should include next/previous buttons, page count, and rows per page options.  All of these are available in the `Pagination` component under `@woocommerce/components`.

### PR DESCRIPTION
**Pull request by mdperez86**
### All Submissions:

-   [x] Have you followed the [WooCommerce Contributing guideline](https://github.com/woocommerce/woocommerce/blob/trunk/.github/CONTRIBUTING.md)?
-   [x] Does your code follow the [WordPress' coding standards](https://make.wordpress.org/core/handbook/best-practices/coding-standards/)?
-   [x] Have you checked to ensure there aren't other open [Pull Requests](https://github.com/woocommerce/woocommerce/pulls) for the same update/change?

### Changes proposed in this Pull Request:

Closes https://github.com/woocommerce/woocommerce/issues/35788

- [ ] This PR is a very minor change/addition and does not require testing instructions (if checked you can ignore/remove the next section).

### How to test the changes in this Pull Request:

1. Go to `Products -> Add New` page or use and existing variable product
2. Add as much attributes you can so you can generate more than 25 variations (This is the only way to see the pagination in action). Generate more than 25 variations of that product.
3. Then copy the id of your product and then visit `/wp-admin/admin.php?page=wc-admin&path=/product/{product_id}`. Replace the `{product_id}` string with your real id.
4. Under `Options` tab and in the `Variations` section you should see the pagination component at the botton of the list if and only if the product has more than 25 variations and if the items per page is less than the total amount of variations.

<video src="https://user-images.githubusercontent.com/13334210/207671173-d5c0418f-ab45-43cc-9464-7a4f3935189d.mov"></video>

### Other information:

-   [x] Have you added an explanation of what your changes do and why you'd like us to include them?
-   [ ] Have you written new tests for your changes, as applicable?
-   [x] Have you created a changelog file for each project being changed, ie `pnpm --filter=<project> changelog add`?

### FOR PR REVIEWER ONLY:

-   [x] I have reviewed that everything is sanitized/escaped appropriately for any SQL or XSS injection possibilities. I made sure Linting is not ignored or disabled.

### CODE REVIEW

**Comment by mdperez86** [comment]
This PR is blocked by https://github.com/woocommerce/woocommerce/pull/35976

File: plugins/woocommerce-admin/client/products/fields/variations/variations.scss
Comment (joshuatf) [inline plugins/woocommerce-admin/client/products/fields/variations/variations.scss:48]: I know @jarekmorawski hasn't responded yet on whether or not we want to change this for all pagination components, but in the meantime I think it would be okay to update the styling just for variations to the casing in the design.
Screen Shot 2022-12-14 at 11 28 58 AM

**Review by joshuatf — DISMISSED** [review]
Testing well and code looks great!  Left one comment about adding in the text transform we discussed earlier, but pre-approving ✅

File: plugins/woocommerce-admin/client/products/fields/variations/variations.scss
Comment (mdperez86) [reply plugins/woocommerce-admin/client/products/fields/variations/variations.scss:48]: That was solved -> [53b3f20](https://github.com/woocommerce/woocommerce/pull/35979/commits/53b3f20f3829d4aa3bfd8130fbc04902595d9639)

File: plugins/woocommerce-admin/client/products/fields/variations/variations.scss
Comment (joshuatf) [inline plugins/woocommerce-admin/client/products/fields/variations/variations.scss]: One more very minor point I missed before- I think this might be:
```suggestion
		padding: $gap;
```

Which gets the height to `68px` to match the design.

**Review by joshuatf — APPROVED** [review]
This looks great and is testing well on my end!

[I committed one minor style change ](https://github.com/woocommerce/woocommerce/pull/35979#discussion_r1049081223)(feel free to revert if I was incorrect on this or if the styling looks different on your end) so that you can merge without needing to get approval again.

File: plugins/woocommerce-admin/client/products/fields/variations/variations.scss
Comment (mdperez86) [reply plugins/woocommerce-admin/client/products/fields/variations/variations.scss]: Thanks!!!!

**Pull request merged**
